dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -252,7 +249,7 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
             var functionVariables = this.GetNextStepVariables(context.Variables, step);
 
             // Execute the step
-            var functionContext = new SKContext(context.Kernel, functionVariables, context.Skills);
+            var functionContext = context.KernelContext.CreateNewContext(functionVariables, context.Skills);
```

Comment:
I think this make sense also because I don't change a property again after introducing it.

Additionaly (Thoughts):
SKContext in my opinion could just be `ExecutionContext` (mixing some of Mark's idea), so anything in this class is understood as being in the `Context`.

Reply:
I would like to understand, what is general and long-term plan of this feature. Syntax `context.KernelContext.CreateNewContext` looks like this is moving to the wrong direction :)
I still think that we should use Context only for function input/output parameter binding, but not for executing functions from other functions.

It would be nice if:
- Users will invoke functions only through `kernel.RunAsync` - as original goal of this PR.
- Functions can execute other functions directly or there will be some orchestrator which will scan function dependencies, build execution graph and execute all functions in required order.
- SKContext is created only once at the beginning of Kernel execution and passed through all functions as same instance. Functions won't be allowed to create new context by themselves.

This is how dependencies looked previously to call function from other function: Kernel -> Plugin -> Kernel.
This is how it looks now: Kernel -> Plugin -> SKContext -> Kernel.

In my opinion, the dependencies should look like: Kernel -> Plugin1 -> Plugin2 -> Plugin3.

Reply:
> SKContext is created only once at the beginning of Kernel execution and passed through all functions as same instance. Functions won't be allowed to create new context by themselves.

Agree here, removed this method and updated with how this should work.

> Functions can execute other functions directly or there will be some orchestrator which will scan function dependencies, build execution graph and execute all functions in required order.

We have the Planner which is our Function -> Function orchestrator.

> Functions can execute other functions directly

This change is a way of allowing it. KernelExecutionContext is not a fullfledged kernel, it's intent is to live only during the execution of the function as a security boudary to what a function also has access.